### ISSUE TITLE
[lc_ctrl / flash / otp] nvm_debug_en availability

### ISSUE THREAD

**Issue by tjaychen** [lowRISC/opentitan#4526]
Right now, `lc_ctrl` exposes an nvm_debug_en signal that is available during test and rma states. 
Due to the need to split up secrets during test states however, we need to be able to remove at least flash debug enable before we completely transition to production states. 

I think ideally it would be nice to pinpoint exactly which state needs this change, but we won't know that information for some time.
So it seems like for now it would be best to add another `HW_CFG` that allows a manufacturer disable flash debug.

The flash wrapper / controller will then take a combination of the life cycle signal and the new config bit to determine whether debug is allowed.

Does this sound reasonable? Or do we think there might be a different / better approach?

**Comment by tjaychen** [comment]
just to chime in on this one again.  I think given some of the features available in `lc_ctrl`, it's fine to just pick a state that disable nvm_debug_en. 

right now it looks like the following 

```
      unique case (lc_state_i)
        ///////////////////////////////////////////////////////////////////
        // Enable DFT and debug functionality, including the CPU in the
        // test unlocked states.
        LcStTestUnlocked0,
        LcStTestUnlocked1,
        LcStTestUnlocked2,
        LcStTestUnlocked3: begin
          lc_dft_en            = On;
          lc_nvm_debug_en      = On;
          lc_hw_debug_en       = On;
          lc_cpu_en            = On;
          lc_iso_part_sw_wr_en = On;
          lc_keymgr_div_d      = RndCnstLcKeymgrDivTestDevRma;
        end
```

I'm thinking we just tweak this slightly such that in 2/3 nvm_debug_en is no longer on. 

let me know what you think.

**Comment by msfschaffner** [comment]
From a design perspective this is easy to do. We just have to agree which unlocked states should not enable NVM debug (probably the last unlocked state?).

**Comment by zi-v** [comment]
Yes. The last.

**Comment by msfschaffner** [comment]
This has been addressed in #7328.
